android/KMEA/app/src/main/java/com/keyman/engine/util/KMLog.java

```diff
@@ -34,6 +35,44 @@ public static void LogInfo(String tag, String msg) {
     }
   }
 
+  /**
+   * Utility to log info and add as a Sentry breadcrumb, rather than
+   * as an independent message
+   * @param tag String of the caller
+   * @param msg String of the info message
+   */
+  public static void LogBreadcrumb(String tag, String msg, boolean addStackTrace) {
+    if (msg != null && !msg.isEmpty()) {
+      Log.i(tag, msg);
+
+      if (DependencyUtil.libraryExists(LibraryType.SENTRY) && Sentry.isEnabled()) {
```

Comment:
nit, but it's better for preconditions to exit early to reduce brace depth:

```suggestion
    if (msg == null || msg.isEmpty()) {
      return;
    }
    
    Log.i(tag, msg);

    if (!DependencyUtil.libraryExists(LibraryType.SENTRY) || !Sentry.isEnabled()) {
      return;
    }
```